src/_lib/modules/dropdowns/dropdown-menu.directive.ts

```diff
@@ -1,23 +1,43 @@
-import { Directive, HostListener, QueryList, AfterContentInit, ContentChildren, OnDestroy } from '@angular/core';
+import { Directive, HostListener, QueryList, AfterContentInit, ContentChildren, OnDestroy, ElementRef, ChangeDetectorRef } from '@angular/core';
 import { DropdownService } from "./dropdown.service";
 import { DropdownDirective } from "./dropdown.directive";
-import { map } from "rxjs/operators";
+import { filter, map } from "rxjs/operators";
 import { Subscription, Observable, merge } from 'rxjs';
 
 @Directive({
-  selector: '[nwDropdownMenu]',
-  exportAs: 'nw-dropdown-menu'
+    selector: '[nwDropdownMenu]',
+    exportAs: 'nw-dropdown-menu'
 })
 export class DropdownMenuDirective implements AfterContentInit, OnDestroy {
 
     @ContentChildren(DropdownDirective) nestedDropdowns: QueryList<DropdownDirective>;
 
     private _childrenVisibilityToggledSub: Subscription;
+    private _scrollSub: Subscription;
 
-    constructor(private _service: DropdownService) {}
+    constructor(
+        private _service: DropdownService,
+        private _element: ElementRef,
```

Comment:
I haven't pulled down the branch, but I'm assuming there's no type checking on `nativeElement.querySelector` and `nativeElement.scrollTo` below. It would be best to add the type param of HTMLElement to the ElementRef

Reply:
Sorry Andrew, I meant to change injection to `ElementRef<HTMLElement>`. That way we don't need to cast and we get type checking

Reply:
Ah right.  I've updated the dependency for the menu element, but the `querySelector` function return 'Element' which doesn't have the `offsetTop` property

![image](https://github.com/NewsWhip/style-guide/assets/129736527/b2af5900-864e-4dd1-884a-14a0d6ab0789)

I have to declare HTMLElement type on the active element to get the property 

![image](https://github.com/NewsWhip/style-guide/assets/129736527/3a893eb3-3eb8-4d20-92e3-1a546ae7e243)

Unless I'm missing something?

Reply:
Although it's a bit annoying, I think querySelector accepts a type param so you could have 

`querySelector<HTMLElement>('.active')`

I think this and your casting to `HTMLElement` are essentially the same so I'm happy to go with your preference